### PR TITLE
[POST] /api/students/search

### PR DESCRIPTION
**Pull request by Chaoyuuu**
# Description

See the linked issues for the details.

- If you have other implementation details, write it down here.

### CODE REVIEW

File: Spring-Boot/Spring-Boot-Student/src/test/java/tw/waterball/judgegirl/springboot/student/controllers/it/StudentControllerTest.java [hunk 1]
```diff
@@ -166,6 +166,21 @@ void WhenGetStudentByNonExistingStudentId_ShouldRespondNotFound() throws Excepti
...
+        Student B = new Student("nameB", "b@example.com", "12345678");
+        Student C = new Student("nameC", "c@example.com", "12345678");
+        signUpStudents(A, B, C);
+
+        String[] emails = {"b@example.com", "c@example.com", "d@example.com"};
+        List<StudentView> students = getStudentsByEmail(emails);
+
+        assertEquals(2, students.size());
+        assertStudent(B, students.get(0));
+        assertStudent(C, students.get(1));
```
Comment (Johnny850807) [inline Spring-Boot/Spring-Boot-Student/src/test/java/tw/waterball/judgegirl/springboot/student/controllers/it/StudentControllerTest.java:181]: `studentsShouldHaveEmails(students, "b@example.com", "c@example.com")`

File: Spring-Boot/Spring-Boot-Student/src/main/java/tw/waterball/judgegirl/springboot/student/controllers/StudentController.java [hunk 2]
```diff
@@ -72,6 +74,13 @@ public StudentView getStudentById(@PathVariable Integer studentId, @RequestHeade
                 });
     }
 
+    @PostMapping("/search")
+    public List<StudentView> getStudentsByEmailList(@RequestBody String[] emails) {
+        GetStudentsByEmailListPresenter presenter = new GetStudentsByEmailListPresenter();
+        getStudentsByEmailListUseCase.execute(new GetStudentsByEmailListUseCase.Request(emails), presenter);
```
Comment (Wally5077) [inline Spring-Boot/Spring-Boot-Student/src/main/java/tw/waterball/judgegirl/springboot/student/controllers/StudentController.java:80]: 或許 execute 第一個參數可以放 emails 就好

File: Student-Service/src/main/java/tw/waterball/judgegirl/studentservice/domain/usecases/student/GetStudentsByEmailListUseCase.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    @AllArgsConstructor
+    @NoArgsConstructor
+    public static class Request {
+        public String[] emails;
+    }
+
+    public interface Presenter {
+        void setStudents(List<Student> students);
```
Comment (Johnny850807) [inline Student-Service/src/main/java/tw/waterball/judgegirl/studentservice/domain/usecases/student/GetStudentsByEmailListUseCase.java:47]: `showStudents`

File: Spring-Boot/Spring-Boot-Student/src/test/java/tw/waterball/judgegirl/springboot/student/controllers/it/StudentControllerTest.java [hunk 4]
```diff
@@ -166,6 +166,21 @@ void WhenGetStudentByNonExistingStudentId_ShouldRespondNotFound() throws Excepti
                 .andExpect(status().isNotFound());
     }
 
+    @Test
+    void GivenThreeStudents_A_B_C_SignedUp_WhenGetStudents_B_C_D_ByEmail_ShouldRespond_B_C() throws Exception {
```
Comment (Johnny850807) [inline Spring-Boot/Spring-Boot-Student/src/test/java/tw/waterball/judgegirl/springboot/student/controllers/it/StudentControllerTest.java:170]: `GivenThreeStudents_A_B_C_SignedUp_WhenGetStudents_B_C_D_ByEmails_ShouldRespond_B_C()`

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Review done

File: Student-Service/src/main/java/tw/waterball/judgegirl/studentservice/domain/usecases/student/GetStudentsByEmailListUseCase.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+import tw.waterball.judgegirl.studentservice.domain.repositories.StudentRepository;
+
+import javax.inject.Named;
+import java.util.List;
+
+/**
+ * @author chaoyulee chaoyu2330@gmail.com
+ */
+@Named
+public class GetStudentsByEmailListUseCase {
```
Comment (Wally5077) [inline Student-Service/src/main/java/tw/waterball/judgegirl/studentservice/domain/usecases/student/GetStudentsByEmailListUseCase.java:28]: 或許可加上 @AllArgsConstructor

**Review by Wally5077 — CHANGES_REQUESTED** [review]
Review done

**Review by Wally5077 — APPROVED** [review]
LGTM

**Review by Johnny850807 — APPROVED** [review]
LGTM

**Pull request merged**